witness/src/Dscp/Witness/CLI.hs

```diff
@@ -85,13 +91,20 @@ metricsServerParser =
 -- Utils
 ---------------------------------------------------------------------------
 
-committeeParamsParser :: Parser CommitteeParams
-committeeParamsParser =
-    combine <$> nParser <*> optional commSecretParser
+committeeKeyParamsParser :: OptModParser CommitteeKeyParams
+committeeKeyParamsParser = #params .:+
+    (#participantN .:: nParser <*<
+     (#paramsType .:: ("open" <$ openP)) <|>
+     ((#paramsType .:: pure "closed" <*<
```

Comment:
This pattern seems to be repeated in CLI parsers for sum types, and it's not particularly pretty. Can we make a helper which does the same thing?

Reply:
Also, why not just do `#paramsType .:: ("open" <$ openP <|> pure "closed")` and parse `#closed` branch in each case? `OptModParser`s never fail anyway, and in case of `open` committee redundant branch will be discarded during config finalization. Basically, you do exactly this in `witnessKeyParamsParser`

Reply:
we cannot do that, because even tho in practice it could work, this would make it so that, in this case, `"closed"` is selected even when we do not specify **any** CLI arguments.

This does in fact break the proposition that we test for, in this case:
>spec_WitnessCliParamsNoDefaults :: Spec
>spec_WitnessCliParamsNoDefaults = describe "Witness node CLI interface" $
>    it "should not yield any default values if no CLI params are provided" $
>        (runCliArgs witnessConfigParser [] <*> pure mempty) \`shouldBe\` Just mempty

does not hold anymore.

In the case of `witnessKeyParamsParser` there is a difference: both the selection possibilities have their own CLI argument, and I think it is necessary here.
Basically for `committeeKeyParamsParser` I was trying to avoid to add another CLI argument just to make the "closed" selection and let the `--comm-sec` option handle that as well, so one can define the content of the `closed` branch and select it at the same time. 

Agreed on the ugliness and the helper idea, I'll try to make one.